src/CMakeLists.txt

```diff
@@ -153,20 +72,15 @@ else ()
     VERBATIM)
 endif (WIN32)
 
-add_library(${PROJECT_NAME} MODULE ${SOURCES} ${TEST_SOURCES} ${GITCOMMIT_H})
-
-if(CMAKE_BUILD_TYPE MATCHES Debug)
-  add_subdirectory(../test/src "${CMAKE_CURRENT_BINARY_DIR}/test/src")
-endif(CMAKE_BUILD_TYPE MATCHES Debug)
+if (CMAKE_BUILD_TYPE MATCHES Debug)
+add_library(${PROJECT_NAME} MODULE ${SOURCES} ${GITCOMMIT_H} $<TARGET_OBJECTS:${PROJECT_NAME}-tests>)
```

Comment:
shouldn't `$<TARGET_OBJECTS:${PROJECT_NAME}-tests>` be `$<TARGET_OBJECTS:${TESTS_LIB_NAME}>`